product_standard_price_restriction/models/product_template.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.exceptions import UserError
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    standard_price = fields.Float(track_visibility='onchange')
+    update_cost = fields.Boolean()
+
+    def write(self, vals):
+        if "standard_price" in vals:
+            if not vals.get('update_cost', self.update_cost):
+                raise UserError(
+                    _(
+                        "You are about to update the cost. This field is "
+                        "auto-calculated by Odoo and is an 'Weighted Average Value' of "
```

Comment:
It is not good to hardcode a message like this. Can we make the message more generic, or put it in a configuration field (upon adding a field perhaps in res.company)?